### PR TITLE
End to end tests for the Guinier apps

### PR DESCRIPTION
**Pull request by maaeli**
Hi,

I have added some end to end tests for the Guinier apps. 
I also changed the formatting string for the RG tuple, as it sometimes inserted additional whitespaces which made parsing trickier...

Cheers,
Martha

### CODE REVIEW

File: e2etest/e2etest_guinier_apps.py [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+                test_output_result[1][0],
+            },
+            msg=f"native file created by {app_name} contains "
+            "the filenames at the start of each row",
+        )
+
+
+def suite():
+    test_suite = unittest.TestSuite()
+    for app in ["auto_gpapy", "auto_guinierpy", "autorgpy"]:
```
Comment (kif) [inline e2etest/e2etest_guinier_apps.py:349]: What about:
```
import sys
if sys.plateform == "win32":
    app = app[:-2]
```

**Comment by kif** [comment]
I agree the trailing empty lines should be considered as abug. 
On the other hand, stripping for empty lines should not be a big deal.

File: e2etest/e2etest_guinier_apps.py (hunk 1 above)
Comment (maaeli) [reply e2etest/e2etest_guinier_apps.py:349]: If I just manually run autorg.py from the powershell it actually works... 
For the tests, using subprocess, using autorg.py.exe works without having to touch the entry points, so I would suggest that solution. The combination of powershell + subprocess is a bit headache inducing...

Comment (maaeli) [reply e2etest/e2etest_guinier_apps.py]: On appveyor  autorg.py.exe becomes autorg.exe. I don't understand why.

**Comment by maaeli** [comment]
I decided to rename all apps with a "." in the name as apparently this is inviting trouble on Windows.

**Comment by kif** [comment]
That's a big change which is likely to break many pipelines ...
Since this is not a version 1.0, it is possible but it is likely to disturb every single user even if I guess there are few of them. 

I am not against the change, but then it would be nice to make all application names consistent.

**Comment by maaeli** [comment]
How about this: All os have point-free script names. These are tested and described in the documentation.
On non-windows, the point-names continue to exist as to not break any pipeline. They eventually could get deprecation warnings.

**Comment by kif** [comment]
Looks great !

**Comment by kif** [comment]
Thanks

**Pull request merged**